addon/model-data.js

```diff
@@ -51,6 +68,12 @@ class M3SchemaInterface {
     return this._refKeyDepkeyMap[refKey];
   }
 
+  /**
+   * Get the attribute of the model data plus recording the key being resolved
```

Comment:
Since this is public API I think it makes more sense to talk about server payloads than model data, which is a concept for addon authors and internals, but not really for app developers.

Maybe something like

"Get the attribute `name` from the model.  This can be useful if your payload keys are different from your attribute names; for example if your api adds a prefix to attributes that should be interpreted as references."